src/pages/VideoDetail/VideoDetail.page.jsx

```diff
@@ -12,20 +12,91 @@ import {
   VideoDescription,
   AddFavorites,
 } from './styled';
-import ThemeContext from '../../state/ThemeContext';
+import GlobalContext from '../../state/GlobalContext';
 
-function VideoDetailPage({ videos }) {
+function VideoDetailPage() {
   const isLogged = storage.get(AUTH_STORAGE_KEY);
   const history = useHistory();
   const [video, setVideo] = React.useState(history.location.state.video);
-  const [vids, setVideos] = React.useState(videos);
+  const [vids, setVideos] = React.useState(history.location.state.videos);
+  const [favorite, setFavorite] = React.useState(false);
   const { snippet, id } = video;
-  const { stateTheme } = useContext(ThemeContext);
-  const { theme } = stateTheme;
-  const addFavorites = <AddFavorites theme={theme}>AGREGAR A FAVORITOS</AddFavorites>;
+  const { state } = useContext(GlobalContext);
+  const { theme } = state;
+  let favoritesList = storage.get('favoriteList');
+
+  const addToFavorites = () => {
+    if (favoritesList === null) {
+      favoritesList = [];
+    }
+    const index = favoritesList.findIndex(
+      (element) => element.id.videoId === video.id.videoId
+    );
+    if (index === -1) {
+      favoritesList.push(video);
+    }
+    storage.set('favoriteList', favoritesList);
+    setFavorite(true);
+  };
+
+  const removeFromFavorites = () => {
+    const index = favoritesList.findIndex(
+      (element) => element.id.videoId === video.id.videoId
+    );
+    if (index !== -1) {
+      favoritesList.splice(index, 1);
+      storage.set('favoriteList', favoritesList);
+    }
+    setFavorite(false);
+  };
+
+  const addFavorites = (
+    <AddFavorites
+      theme={theme}
+      onClick={() => {
+        addToFavorites();
+      }}
+    >
+      AGREGAR A FAVORITOS
+    </AddFavorites>
+  );
+
+  const removeFavorites = (
+    <AddFavorites
+      theme={theme}
+      onClick={() => {
+        removeFromFavorites();
+      }}
+    >
+      REMOVER DE FAVORITOS
+    </AddFavorites>
+  );
+
+  const isFavorite = () => {
+    if (favoritesList !== null) {
+      const index = favoritesList.findIndex(
+        (element) => element.id.videoId === video.id.videoId
+      );
+      if (index !== -1) {
+        setFavorite(true);
+      } else {
+        setFavorite(false);
+      }
```

Comment:
could be reduced to setFavorite()
```suggestion
        setFavorite(index !== -1);
```